frontend/src/components/journal.js

```diff
@@ -1,11 +1,22 @@
 import React, { useState, useEffect } from 'react';
-import axios from 'axios';
+import { Grid, Card, RingProgress, Text, Center } from '@mantine/core';
+import { BiRun, BiCycling, BiDumbbell, BiSwim } from 'react-icons/bi'; // Import icons for exercise types
+import { FaPersonCircleQuestion } from "react-icons/fa6";
 import { Button } from 'react-bootstrap';
 import moment from 'moment';
+import axios from 'axios';
 import './journal.css';
 import config from '../config';
 
-const Journal = ({ currentUser }) => {
+const iconMap = {
+    Running: { icon: BiRun, color: '#322142' },
+    Cycling: { icon: BiCycling, color: '#03A6A6' },
+    Gym: { icon: BiDumbbell, color: '#00563E' },
+    Swimming: { icon: BiSwim, color: '#CE466B' },
+    Other: { icon: FaPersonCircleQuestion, color: '#F25757' } // Changed icon to FaUserCircle
+};
+
+const Journal = ({ currentUser, weeklyGoal=120 }) => {
```

Comment:
I've started a discussion regarding weekly goals in the group chat, but this 120min looks like a great starting point.